### PR TITLE
Relax assertion in `InitMilestoneTest`

### PR DESCRIPTION
**Pull request by basil**
This test, while not extremely flaky, is one of the more consistent flakes I have noticed, so dealing with it is a prerequisite to turning off `surefire.rerunFailingTestsCount`. However I am completely stumped about the root cause of the problem. While we always get all the entries we expect, the ordering is usually but not always consistent. I doubt this is actually a problem in production, since if Jenkins really couldn't start properly I think we would have heard about it by now. Rather I suspect the test is just making some false assumption about how the test extensions are executed. Since this needs to be dealt with sooner rather than later, I think the path of least resistance is to relax the test for now, since it doesn't seem to be adding much value in the way of detecting bugs and regressions. If someone wants to spend a lot more time debugging the cause of the flakiness, we can always improve it in the future. For now I am just asserting that the expected elements are there in any order.

### Testing done

I've looped this for over 5 hours without a flake. In contrast, every other attempt I have tried at a fix has flaked within about 2 hours.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  - Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7188"></a>

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: test/src/test/java/hudson/init/InitMilestoneTest.java
Comment (basil) [reply test/src/test/java/hudson/init/InitMilestoneTest.java:25]: Good point

**Review by NotMyFault — APPROVED** [review]
This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Comment by basil** [comment]
Trying to put this fire out by the end of the week…

**Pull request merged**
